### ISSUE TITLE
Analyser memory leak

### ISSUE THREAD

**Issue by CodeRunRepeat** [microsoft/presidio#198]
Run presidio under persistent load for a long time. 
notice memory in analyser increases over time, eventually the pod will restart.

ruled out recognizers, context and other factors by marking out all code in analyzer's main methods (Apply and analyze) except for nlp_engine.process_text() and memory still increases.

this is aligned with the following github issues on spacy memory increase over time, due to internal cache usage:

https://github.com/explosion/spaCy/issues/3618
https://github.com/explosion/spaCy/issues/3013
https://github.com/explosion/spaCy/issues/285

**Comment by honnibal** [comment]
If you get really stuck, one solution would be to just reload the `nlp` object periodically.

How much is the memory increase, over how much text?

**Comment by omri374** [comment]
Hi @honnibal, thanks for your comment!
We did a soak test on one Dv3 vm on Azure (https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes-general#dv3-series-1)
We ran the soak test for 12 hours, with a request with 100 words, ~40 times a second.

So in total we ingested 1.7 million requests = 172 million words. At that point the ram usage was at 6gb which was the maximum allocated for that container.
Memory increase was from 2gb (initial memory usage) to 6gb over the course of 12 hours.

Keep up the good work with Spacy!

cc @balteravishay
